### PR TITLE
Introspection: store discontinuous per composition

### PR DESCRIPTION
**Pull request by tjhei**
This changes the type of ``use_discontinuous_composition_discretization`` to a std::vector<bool> and replaces all usages of this variable. No other new/different functionality.

part of #5748

### CODE REVIEW

File: source/simulator/core.cc [hunk 1]
```diff
@@ -720,13 +720,13 @@ namespace aspect
...
-        // composition boundary conditions and interpolate the composition
-        // there
-        for (unsigned int c=0; c<introspection.n_compositional_fields; ++c)
+    // now do the same for the composition variables:
+    {
+      // obtain the boundary indicators that belong to Dirichlet-type
+      // composition boundary conditions and interpolate the composition
+      // there
+      for (unsigned int c=0; c<introspection.n_compositional_fields; ++c)
+        if (parameters.use_discontinuous_composition_discretization[c])
```
Comment (gassmoeller) [inline source/simulator/core.cc:729]: I think in order to be the same as before this needs to be:
```suggestion
        if (parameters.use_discontinuous_composition_discretization[c] == false)
```

File: unit_tests/introspection.cc [hunk 2]
```diff
@@ -24,7 +24,7 @@
 #include <aspect/simulator.h>
 #include <aspect/introspection.h>
 
-TEST_CASE("Introspection::1")
```
Comment (gassmoeller) [inline unit_tests/introspection.cc:27]: why this change?

**Review by gassmoeller — COMMENTED** [review]
Looks ok to me, but it seems to change some tests. Check the place that I marked, maybe that is the reason. Do I understand correctly that with this PR users cannot input a vector (because it is not declared as a vector), but internally it is already handled as a vector, just containing identical entries? If so this is good to merge once you figure out the reason for the test failures.

File: source/simulator/core.cc (hunk 1 above)
Comment (tjhei) [reply source/simulator/core.cc]: Thank you!

**Comment by tjhei** [comment]
updated.

**Review by gassmoeller — APPROVED** [review]
